src/lib/quests/baseQuest.ts

```diff
@@ -0,0 +1,40 @@
+import MatchStatistics, { isPlayerInMatchFilter } from "../matchStatistics";
+import { type Quest, type QuestStatus, type QuestType } from "../quest";
+
+export abstract class BaseQuest<TConditionData, TState>
+  implements Quest<TConditionData, TState>
+{
+  abstract state: TState;
```

Comment:
Behøver BaseQuest kende til state? Er det ikke en implementeringsdetalje som hver enkel quest kan bruge eller lade være?

Hvis BaseQuest ikke har state så er vi frie til at kalde den lige hvad vi vil, fx kan `PlayMatchCountQuest` have propertien `matchCount`, hvor `Play1v1Quest` kan have `hasPlayed1v1` from for at de begge skal hedde `state`